processing.go

```diff
@@ -24,6 +26,50 @@ type ScanTarget struct {
 	Domain string
 }
 
+// scanTargetConnection wraps an existing net.Conn connection, overriding the Read/Write methods to use the configured timeouts
+type scanTargetConnection struct {
+	net.Conn
+	Timeout time.Duration
+}
+
+func (c *scanTargetConnection) Read(b []byte) (n int, err error) {
```

Comment:
My main concern here is that some scanners may need more flexibility -- say, different timeouts for different reads.

But, my thinking is that in those cases, the scanner is going to need to know about the timeout anyway, so the it can go ahead and open its own connection with net.Dial directly.